### PR TITLE
Feature/SK-510 | Create clients using studio-cli

### PR DESCRIPTION
**Pull request by niklastheman**
List and create clients

### CODE REVIEW

File: stackn/create.py [hunk 1]
```diff
@@ -168,6 +169,17 @@ def project(name, description, template, studio_url, secure):
...
 
 
+@create.command("client")
+@click.argument("name")
+@click.option("-p", "--project", required=False, default=[])
+@click.option("-c", "--controller-id", required=False, default=[])
+@click.option("-r", "--role", required=False, default=[])
+@click.option("-u", "--studio-url", required=False, default=[])
+@click.option("--secure/--insecure", default=True)
+def client(name, project, controller_id, role, studio_url, secure):
```
Comment (Wrede) [inline stackn/create.py:179]: add doc string, I know many of the other functions lack it but we should start adding it. Also add click help text.

File: stackn/get.py [hunk 2]
```diff
@@ -348,6 +348,38 @@ def s3(project, studio_url, name, secure):
         )
 
 
+@get.command("clients")
+@click.option("-p", "--project", required=False, default=[])
+@click.option("-u", "--studio-url", required=False, default=[])
+@click.option("--secure/--insecure", required=False, default=True)
+def clients(project, studio_url, secure):
```
Comment (Wrede) [inline stackn/get.py:355]: doc sting and help text

File: stackn/stackn.py [hunk 3]
```diff
@@ -817,6 +819,113 @@ def delete_meta_resource(
         print(res.text)
 
 
+def create_client(
```
Comment (Wrede) [inline stackn/stackn.py:822]: add doc string

File: stackn/create.py [hunk 4]
```diff
@@ -168,6 +169,17 @@ def project(name, description, template, studio_url, secure):
...
+    create_client(name, project, controller_id, role, studio_url, secure)
```
Comment (Wrede) [inline stackn/create.py:189]: function can return False, if the value is not used do: _ = create_client(**args)

File: stackn/stackn.py [hunk 5]
```diff
@@ -817,6 +819,113 @@ def delete_meta_resource(
...
+        return False
+
+    data = {"name": name, "controller": controller_id, "role": role}
+
+    res = requests.post(
+        url, headers=auth_header, json=data, verify=conf["STACKN_SECURE"]
+    )
+
+    if res:
+        print(res.text)
```
Comment (Wrede) [inline stackn/stackn.py:937]: I would also return True if function was successful, to keep it consistent with error handling by returning False

File: stackn/stackn.py [hunk 6]
```diff
@@ -817,6 +819,113 @@ def delete_meta_resource(
...
+    else:
+        print("Failed to create client.")
+        print("Status code: {}".format(res.status_code))
+        print(res.text)
```
Comment (Wrede) [inline stackn/stackn.py:941]: return False

**Review by Wrede — CHANGES_REQUESTED** [review]

File: stackn/stackn.py (hunk 3 above)
Comment (niklastheman) [reply stackn/stackn.py:822]: **See image:**

Screenshot 2023-06-28 at 14 08 45

File: stackn/get.py (hunk 2 above)
Comment (niklastheman) [reply stackn/get.py:355]: **See image:**

Screenshot 2023-06-28 at 14 08 45

File: stackn/create.py (hunk 1 above)
Comment (niklastheman) [reply stackn/create.py:179]: done

File: stackn/stackn.py (hunk 3 above)
Comment (niklastheman) [reply stackn/stackn.py:822]: done

File: stackn/create.py (hunk 4 above)
Comment (niklastheman) [reply stackn/create.py:189]: Updated so that function never returns a value

File: stackn/stackn.py (hunk 5 above)
Comment (niklastheman) [reply stackn/stackn.py:937]: Updated so that function never returns a value

File: stackn/stackn.py (hunk 6 above)
Comment (niklastheman) [reply stackn/stackn.py:941]: Updated so that function never returns a value

**Review by Wrede — APPROVED** [review]

**Pull request merged**